pkg/deploy/api/register.go

```diff
@@ -1,47 +1,83 @@
 package api
 
 import (
+	kapi "k8s.io/kubernetes/pkg/api"
 	"k8s.io/kubernetes/pkg/api/unversioned"
+	"k8s.io/kubernetes/pkg/apis/extensions"
 	"k8s.io/kubernetes/pkg/runtime"
+	"k8s.io/kubernetes/pkg/watch/versioned"
 )
 
-const GroupName = ""
-const FutureGroupName = "deploy.openshift.io"
+const (
+	LegacyGroupName = ""
+	GroupName       = "apps.openshift.io"
+)
+
+var (
+	SchemeGroupVersion       = unversioned.GroupVersion{Group: GroupName, Version: runtime.APIVersionInternal}
+	LegacySchemeGroupVersion = unversioned.GroupVersion{Group: LegacyGroupName, Version: runtime.APIVersionInternal}
+
+	LegacySchemeBuilder    = runtime.NewSchemeBuilder(addLegacyKnownTypes)
+	AddToSchemeInCoreGroup = LegacySchemeBuilder.AddToScheme
 
-// SchemeGroupVersion is group version used to register these objects
-var SchemeGroupVersion = unversioned.GroupVersion{Group: GroupName, Version: runtime.APIVersionInternal}
+	SchemeBuilder = runtime.NewSchemeBuilder(addKnownTypes)
+	AddToScheme   = SchemeBuilder.AddToScheme
+)
 
 // Kind takes an unqualified kind and returns back a Group qualified GroupKind
 func Kind(kind string) unversioned.GroupKind {
 	return SchemeGroupVersion.WithKind(kind).GroupKind()
 }
 
+// LegacyKind takes an unqualified kind and returns back a Group qualified GroupKind
+func LegacyKind(kind string) unversioned.GroupKind {
+	return LegacySchemeGroupVersion.WithKind(kind).GroupKind()
+}
+
 // Resource takes an unqualified resource and returns back a Group qualified GroupResource
 func Resource(resource string) unversioned.GroupResource {
 	return SchemeGroupVersion.WithResource(resource).GroupResource()
 }
 
-var (
-	SchemeBuilder = runtime.NewSchemeBuilder(addKnownTypes)
-	AddToScheme   = SchemeBuilder.AddToScheme
-)
+// LegacyResource takes an unqualified resource and returns back a Group qualified GroupResource
+func LegacyResource(resource string) unversioned.GroupResource {
+	return LegacySchemeGroupVersion.WithResource(resource).GroupResource()
+}
+
+// Adds the list of known types to api.Scheme.
+func addLegacyKnownTypes(scheme *runtime.Scheme) error {
+	types := []runtime.Object{
+		&DeploymentConfig{},
+		&DeploymentConfigList{},
+		&DeploymentConfigRollback{},
+		&DeploymentRequest{},
+		&DeploymentLog{},
+		&DeploymentLogOptions{},
+		&extensions.Scale{},
```

Comment:
this can be removed from the legacy scheme I believe... however if I remove it from the api group the server won't start.